ads/google/utils.js

```diff
@@ -109,35 +125,41 @@ export function getMultiSizeDimensions(
  *
  * @param {(number|string)} width
  * @param {(number|string)} height
- * @param {!function((number|string)): boolean} widthCond
- * @param {!function((number|string)): boolean} heightCond
- * @param {!function(!{badDim: string, badVal: (string|number)}): string}
- *    errorBuilder A function that will produce an informative error message.
+ * @param {function((number|string)): boolean} widthCond
+ * @param {function((number|string)): boolean} heightCond
+ * @param {function(!{badDim: string, badVal: (string|number)}): string=}
+ *   errorBuilder A function that will produce an informative error message.
+ * @param {boolean=} reportError If true, indicates that an error message
+ *   should be logged to the console.
  * @return {boolean}
  */
 function validateDimensions(width, height, widthCond, heightCond,
-    errorBuilder) {
+    errorBuilder, reportError) {
   let badParams = null;
-  if (widthCond(width) && heightCond(height)) {
+  const isWidthIllegal = widthCond(width);
+  const isHeightIllegal = heightCond(height);
+  if (isWidthIllegal && isHeightIllegal) {
```

Comment:
This seems overally verbose.  I would instead allow badParams to be an array and errorBuilder should handle it as such:

if (widthCond(width)) { badParams.push({...}); }
if (heightCond(height)) { badParams.push({...}); }
user().assert(!reportParam || !badParams.length, errorBuilder(badParams));
return !badParams.length;